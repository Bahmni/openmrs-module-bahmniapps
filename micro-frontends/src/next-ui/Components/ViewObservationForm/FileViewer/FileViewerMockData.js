export const mockInfoMap = {
  providerName: "test provider",
  encounterDateTime: "encounterDateTime",
  comment: "comment",
  fileName: "fileName",
  fileType: "image/jpeg",
  fileSource: "fileSource",
};

export const mockVideoInfoMap = {
  providerName: "test provider",
  encounterDateTime: "encounterDateTime",
  comment: "comment",
  fileName: "fileName",
  fileType: "image/jpeg",
  fileSource: "fileSource",
};

export const mockIndex = 0;

export const mockItem = {
  providers: [
    {
      name: "test provider",
    },
  ],
  encounterDateTime: 1713856719000,
  comment: "comment",
  complexData: {
    title: "fileName",
    mimeType: "image/jpeg",
  },
  value: "fileSource",
};

export const mockPdfItem = {
  providers: [
    {
      name: "test provider",
    },
  ],
  encounterDateTime: 1713856719000,
  comment: "comment",
  complexData: {
    title: "fileName",
    mimeType: "application/pdf",
  },
  value: "fileSource",
};

export const mockVideoItem = {
  providers: [
    {
      name: "test provider",
    },
  ],
  encounterDateTime: 1713856719000,
  comment: "comment",
  complexData: {
    title: "fileName",
    mimeType: "video/mp4",
  },
  value: "fileSource",
};

export const observationList = [
    {
      "encounterDateTime": 1713856719000,
      "groupMembers": [],
      "providers": [
        {
          "uuid": "c1c26908-3f10-11e4-adec-0800271c1b75",
          "name": "test provider",
          "encounterRoleUuid": "a0b03050-c99b-11e0-9572-0800200c9a66"
        }
      ],
      "isAbnormal": null,
      "duration": null,
      "type": "Complex",
      "encounterUuid": "276f3f45-2bdb-4166-b495-8e3463774034",
      "interpretation": null,
      "complexData": {
        "title": "51196-Consultation-3c01db16-e63a-4b83-9159-dc3f4e8ebb01.mp4",
        "mimeType": "video/mp4",
      },
      "conceptNameToDisplay": "Patient Video",
      "concept": {
        "uuid": "eef8e969-269d-40c4-b7c9-182da11c2653",
        "name": "Patient Video",
        "dataType": "Complex",
        "shortName": "Patient Video",
      },
      "conceptUuid": "eef8e969-269d-40c4-b7c9-182da11c2653",
      "comment": "Sample comments are added",
      "value": "51200/51196-Consultation-3c01db16-e63a-4b83-9159-dc3f4e8ebb01.mp4"
    },
    {
      "encounterDateTime": 1713856719000,
      "groupMembers": [],
      "providers": [
        {
          "uuid": "c1c26908-3f10-11e4-adec-0800271c1b75",
          "name": "test provider",
        }
      ],
      "isAbnormal": null,
      "type": "Complex",
      "interpretation": null,
      "complexData": {
        "title": "51196-Consultation-37e2a799-b44c-412f-9a16-1f04ae8f9a34.mp4",
        "mimeType": "video/mp4",
      },
      "conceptNameToDisplay": "Patient Video",
      "concept": {
        "uuid": "eef8e969-269d-40c4-b7c9-182da11c2653",
        "name": "Patient Video",
        "dataType": "Complex",
        "shortName": "Patient Video",
      },
      "conceptUuid": "eef8e969-269d-40c4-b7c9-182da11c2653",
      "comment": "Sample comments are added",
      "value": "51200/51196-Consultation-37e2a799-b44c-412f-9a16-1f04ae8f9a34.mp4"
    },
    {
      "encounterDateTime": 1713856719000,
      "groupMembers": [],
      "providers": [
        {
          "uuid": "c1c26908-3f10-11e4-adec-0800271c1b75",
          "name": "test provider",
        }
      ],
      "isAbnormal": null,
      "type": "Complex",
      "interpretation": null,
      "complexData": {
        "title": "51196-Consultation-b1eae63b-98a0-4396-80c7-0beb06f5fd00.jpeg",
        "mimeType": "image/jpeg",
      },
      "conceptNameToDisplay": "Image",
      "concept": {
        "uuid": "6ca8ddf3-4e3d-4116-bd4e-a5147e580931",
        "name": "Image",
        "dataType": "Complex",
        "shortName": "Image",
      },
      "comment": "Sample comments are added",
      "value": "51200/51196-Consultation-b1eae63b-98a0-4396-80c7-0beb06f5fd00.jpeg"
    },
    {
      "encounterDateTime": 1713856719000,
      "groupMembers": [],
      "providers": [
        {
          "uuid": "c1c26908-3f10-11e4-adec-0800271c1b75",
          "name": "test provider",
          "encounterRoleUuid": "a0b03050-c99b-11e0-9572-0800200c9a66"
        }
      ],
      "isAbnormal": null,
      "type": "Complex",
      "encounterUuid": "276f3f45-2bdb-4166-b495-8e3463774034",
      "interpretation": null,
      "complexData": {
        "title": "51196-Consultation-445dd7fc-a7c7-4e9c-8ea6-70a3f567d8d4.jpeg",
        "mimeType": "image/jpeg",
      },
      "conceptNameToDisplay": "Image",
      "concept": {
        "uuid": "6ca8ddf3-4e3d-4116-bd4e-a5147e580931",
        "name": "Image",
        "dataType": "Complex",
        "shortName": "Image",
      },
      "comment": "Sample comments are added",
      "value": "51200/51196-Consultation-445dd7fc-a7c7-4e9c-8ea6-70a3f567d8d4.jpeg"
    },
    {
      "encounterDateTime": 1713856719000,
      "groupMembers": [],
      "providers": [
        {
          "uuid": "c1c26908-3f10-11e4-adec-0800271c1b75",
          "name": "test provider",
        }
      ],
      "isAbnormal": null,
      "type": "Complex",
      "interpretation": null,
      "complexData": {
        "title": "51196-Consultation-46b1cbfc-00a7-4cbb-afdf-71ac2b040182.jpeg",
        "mimeType": "image/jpeg",
      },
      "conceptNameToDisplay": "Image",
      "concept": {
        "uuid": "6ca8ddf3-4e3d-4116-bd4e-a5147e580931",
        "name": "Image",
        "dataType": "Complex",
        "shortName": "Image",
      },
      "conceptUuid": "6ca8ddf3-4e3d-4116-bd4e-a5147e580931",
      "comment": "Sample comments are added",
      "value": "51200/51196-Consultation-46b1cbfc-00a7-4cbb-afdf-71ac2b040182.jpeg"
    },
    {
      "encounterDateTime": 1713856719000,
      "visitStartDateTime": null,
      "targetObsRelation": null,
      "groupMembers": [],
      "providers": [
        {
          "uuid": "c1c26908-3f10-11e4-adec-0800271c1b75",
          "name": "test provider",
          "encounterRoleUuid": "a0b03050-c99b-11e0-9572-0800200c9a66"
        }
      ],
      "isAbnormal": null,
      "duration": null,
      "type": "Complex",
      "encounterUuid": "276f3f45-2bdb-4166-b495-8e3463774034",
      "obsGroupUuid": null,
      "creatorName": "test provider",
      "conceptSortWeight": 1,
      "parentConceptUuid": null,
      "hiNormal": null,
      "lowNormal": null,
      "formNamespace": "Bahmni",
      "formFieldPath": "Demo Final.1/1-2",
      "interpretation": null,
      "status": "AMENDED",
      "encounterTypeName": null,
      "complexData": {
        "data": null,
        "title": "51196-Consultation-aa0a417a-0a61-4c97-b4be-fe3fc43ea371.pdf",
        "mimeType": "application/pdf",
        "length": null
      },
      "unknown": false,
      "orderUuid": null,
      "abnormal": null,
      "observationDateTime": 1713856755000,
      "conceptNameToDisplay": "Image",
      "valueAsString": "51200/51196-Consultation-aa0a417a-0a61-4c97-b4be-fe3fc43ea371.pdf",
      "concept": {
        "uuid": "6ca8ddf3-4e3d-4116-bd4e-a5147e580931",
        "name": "Image",
        "dataType": "Complex",
        "shortName": "Image",
        "units": null,
        "conceptClass": "Misc",
        "hiNormal": null,
        "lowNormal": null,
        "set": false,
        "mappings": []
      },
      "conceptUuid": "6ca8ddf3-4e3d-4116-bd4e-a5147e580931",
      "voided": false,
      "voidReason": null,
      "uuid": "8a643679-fcb2-4246-9b10-01a3d1eb4eb6",
      "comment": null,
      "value": "51200/51196-Consultation-aa0a417a-0a61-4c97-b4be-fe3fc43ea371.pdf"
    }
  ];