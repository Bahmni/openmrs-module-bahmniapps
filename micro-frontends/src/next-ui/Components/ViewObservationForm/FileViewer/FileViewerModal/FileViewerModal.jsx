import React from "react";
import propTypes from "prop-types";
import { Modal } from "carbon-components-react";
import "./FileViewerModal.scss";

export const FileViewerModal = (props) => {
  const { closeModel, infoMap } = props;
  return (
    <Modal
      open
      passiveModal
      className="file-viewer-modal"
      onRequestClose={closeModel}
    >
      {infoMap.fileType === "video/mp4" ? (
        <video className="file-modal" aria-label="video-tag" controls>
          <source src={infoMap.fileSource} type="video/mp4" />
          Your browser does not support the video tag.
        </video>
      ) : (
        <img src={infoMap.fileSource} alt="image" className="file-modal" />
      )}
      {
        <div className="modal-footer">
          {infoMap?.comment && (
            <div className="modal-comment">{infoMap?.comment}</div>
          )}
          <span>{infoMap?.providerName}</span>&nbsp;
          <span>{infoMap?.encounterDateTime}</span>
        </div>
      }
    </Modal>
  );
};

FileViewerModal.propTypes = {
  closeModel: propTypes.func,
  infoMap: propTypes.object,
};

export default FileViewerModal;
