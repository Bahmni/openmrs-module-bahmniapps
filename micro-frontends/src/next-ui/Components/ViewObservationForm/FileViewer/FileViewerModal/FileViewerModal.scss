.file-viewer-modal {
    width: 100%;
    height: 100%;
}

.file-modal {
    width: 100%;
    height: auto;
}

.modal-comment {
    border: 1px solid #b4b7b9;
    border-radius: 4px;
    padding: 5px;
}