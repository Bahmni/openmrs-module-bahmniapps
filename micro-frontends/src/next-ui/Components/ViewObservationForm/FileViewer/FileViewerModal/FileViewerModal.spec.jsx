import React from "react";
import { render, screen } from "@testing-library/react";
import { FileViewerModal } from "./FileViewerModal.jsx";
import { mockInfoMap, mockVideoInfoMap } from "../FileViewerMockData.js";

describe("FileViewerModal", function () {
  const closeModel = jest.fn();
  
  it("should render FileViewerModal", function () {
    const { container } = render(
      <FileViewerModal closeModel={closeModel} infoMap={mockInfoMap} />
    );
    expect(container).toMatchSnapshot();
  });

  it("should render FileViewerModal with all the data with image", function () {
    const { container } = render(
      <FileViewerModal closeModel={closeModel} infoMap={mockInfoMap} />
    );
    expect(container.querySelector(".file-modal")).not.toBeNull();
    expect(screen.getByAltText(/image/i)).toBeTruthy();
    expect(screen.getByText(mockInfoMap.comment)).toBeTruthy();
    expect(screen.getByText(mockInfoMap.providerName)).toBeTruthy();
    expect(screen.getByText(mockInfoMap.encounterDateTime)).toBeTruthy();
  });

  it("should render FileViewerModal with all the data with video", function () {
    const { container } = render(
      <FileViewerModal closeModel={closeModel} infoMap={mockVideoInfoMap} />
    );
    expect(container.querySelector(".file-modal")).not.toBeNull();
    expect(screen.queryByLabelText("video-tag", { hidden: false } )).toBe(null);
    expect(screen.getByText(mockVideoInfoMap.comment)).toBeTruthy();
    expect(screen.getByText(mockVideoInfoMap.providerName)).toBeTruthy();
    expect(screen.getByText(mockVideoInfoMap.encounterDateTime)).toBeTruthy();
  });
});