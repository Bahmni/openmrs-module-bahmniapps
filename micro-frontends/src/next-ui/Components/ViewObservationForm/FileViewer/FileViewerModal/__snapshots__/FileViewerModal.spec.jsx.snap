// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FileViewerModal should render FileViewerModal 1`] = `
<div>
  <div
    class="bx--modal is-visible file-viewer-modal"
    role="presentation"
  >
    <span
      class="bx--visually-hidden"
      role="link"
      tabindex="0"
    >
      Focus sentinel
    </span>
    <div
      aria-label=""
      aria-modal="true"
      class="bx--modal-container"
      role="dialog"
      tabindex="-1"
    >
      <div
        class="bx--modal-header"
      >
        <button
          aria-label="close"
          class="bx--modal-close"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="bx--modal-close__icon"
            fill="currentColor"
            focusable="false"
            height="20"
            preserveAspectRatio="xMidYMid meet"
            viewBox="0 0 32 32"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M24 9.4L22.6 8 16 14.6 9.4 8 8 9.4 14.6 16 8 22.6 9.4 24 16 17.4 22.6 24 24 22.6 17.4 16 24 9.4z"
            />
          </svg>
        </button>
        
        <h3
          class="bx--modal-header__heading"
          id="bx--modal-header__heading--modal-1"
        />
      </div>
      <div
        aria-labelledby="bx--modal-header__heading--modal-1"
        class="bx--modal-content"
        id="bx--modal-body--modal-1"
      >
        <img
          alt="image"
          class="file-modal"
          src="fileSource"
        />
        <div
          class="modal-footer"
        >
          <div
            class="modal-comment"
          >
            comment
          </div>
          <span>
            test provider
          </span>
          Â 
          <span>
            encounterDateTime
          </span>
        </div>
      </div>
    </div>
    <span
      class="bx--visually-hidden"
      role="link"
      tabindex="0"
    >
      Focus sentinel
    </span>
  </div>
</div>
`;
